subsys/mgmt/mcumgr/lib/smp/include/smp/smp.h

```diff
@@ -25,14 +25,39 @@
 #ifndef H_SMP_
 #define H_SMP_
 
-#include "mgmt/mgmt.h"
```

Comment:
This needs a zcbor and zephyr kernel include

Reply:
No it does not. Nothing in this header defines anything that is `struct net_buf` only `struct net_buf*`.
Definition of `struct net_buf` is only needed on implementation side.

Reply:
This places the onus of file includes on the person that includes this file, the end user should not have to add file includes to their files to be able to include this file without build errors